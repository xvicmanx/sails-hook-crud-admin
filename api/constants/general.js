const SALT_ROUNDS = 10;

module.exports = {
  SALT_ROUNDS,
};
