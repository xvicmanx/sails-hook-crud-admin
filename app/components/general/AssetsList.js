import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Item, Card, Icon, Image } from 'semantic-ui-react';
import Constants from '../../constants';
import { modelTitle } from '../../helpers/models';

const styles = {
  item: {
    fontSize: '1em',
    width: 'fit-content',
    display: 'inline-flex',
    border: '1px solid #ccc',
    margin: '.5rem',
    borderRadius: '5px',
    overflow: 'hidden',
    padding: '1rem',
  },
  picture: {
    width: '100%',
    maxWidth: '200px',
    maxHeight: '200px',
    objectFit: 'contain',
  }
};

const isPicture = type => Constants.ASSETS_TYPES.PICTURE === type;

const AssetItem = ({ type, url, name }) => (
  <Card style={styles.item}>
    {isPicture(type) && (
      <Card.Content>
        <Image
          src={url}
          style={styles.picture}
        />
      </Card.Content>
    )}

    {!isPicture(type) && (
      <Card.Content>
        <Icon
          size="huge"
          name="file outline"
        />
      </Card.Content>
    )}

    <Card.Header textAlign="center">
      {name}
      {!isPicture(type) && (
      <a href={url}>
        &nbsp;&nbsp;<Icon size="large" name="download"/>
      </a>
    )}
    </Card.Header>
  </Card>
);

const AssetsList = props => {
  const { items, type } = props;
  const filteredItems = items.filter(x => x.type === props.type);
  const groupedItems = _.groupBy(filteredItems, 'model');
  const keys = Object.keys(groupedItems);

  return (
    <div>
      {keys.map(name => (
        <Card fluid>
          <Card.Content>
            <Card.Header>
              {modelTitle(name)} {props.type}s
            </Card.Header>
          </Card.Content>
          <Card.Content>
            <Card.Group>
              {groupedItems[name].map(item => (
                <AssetItem
                  {...item}
                  type={props.type}
                />
              ))}
            </Card.Group>
          </Card.Content>
        </Card>
      ))}
    </div>
  );
};

AssetsList.propTypes = {
  items: PropTypes.instanceOf(Array).isRequired,
  type: PropTypes.string.isRequired
};

export default AssetsList;