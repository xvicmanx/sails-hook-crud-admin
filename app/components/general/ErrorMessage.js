import React from 'react';
import PropTypes from 'prop-types';
import { Label } from 'semantic-ui-react';

const ErrorMessage = ({ field, errors, touched }) => {
  if (!touched[field] || !errors[field]) return null;
  return (
    <Label basic color='red' pointing>
      {errors[field]}
    </Label>
  );
};

ErrorMessage.propTypes = {};

export default ErrorMessage;
