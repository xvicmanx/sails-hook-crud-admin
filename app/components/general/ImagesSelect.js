import React from 'react';
import PropTypes from 'prop-types';
import { Card, Divider, Message } from 'semantic-ui-react';

const styles = {
  container: {
    margin: '1rem 0',
  },
  image: {
    width: '100%',
    maxWidth: '120px',
  },
};

const selectionClass = selected => selected && 'selected-picture';

class ImagesSelect extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedImage: null,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ selectedImage: nextProps.value });
  }

  handleImageSelection(selectedImage, evt) {
    const { name, onChange } = this.props;
    this.setState({ selectedImage });
    const event = evt;
    event.target.value = selectedImage;
    event.target.name = name;
    onChange(event);
  }

  render() {
    const { images } = this.props;
    const { selectedImage } = this.state;

    if (images.lenth <= 0) {
      return (
        <Message
          info
          icon="info"
          content="There are no items to show"
        />
      );
    }

    return (
      <div style={styles.container}>
        <Card.Group itemsPerRow={3}>
          {images.map(image => (
            <Card
              onClick={(evt) => {
                this.handleImageSelection(image, evt);
              }}
              raised
              image={image}
              className={selectionClass(selectedImage === image)}
              style={styles.image}
            />
          ))}
        </Card.Group>
        <Divider />
      </div>
    );
  }
}
ImagesSelect.propTypes = {
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

export default ImagesSelect;
