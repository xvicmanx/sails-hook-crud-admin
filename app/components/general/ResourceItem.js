import React from 'react';
import PropTypes from 'prop-types';

class ResourceItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      result: null,
    };
  }

  componentDidMount() {
    this.props.data.then((result) => {
      this.setState({ result });
    });
  }

  render() {
    return (
      <div>
        {this.props.children(
          this.state.result
          || this.props.defaultValue,
        )}
      </div>
    );
  }
}

ResourceItem.propTypes = {
  data: PropTypes.instanceOf(Promise).isRequired,
  children: PropTypes.func.isRequired,
};

export default ResourceItem;
