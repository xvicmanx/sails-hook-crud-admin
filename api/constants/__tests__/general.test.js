import * as constants from '../general';

describe('api::constants::general', () => {
  test('The CRUD model names are correct', () => {
    expect(constants.CRUD_MODELS).toMatchSnapshot();
  });

  test('The HTTP methods are correct', () => {
    expect(constants.HTTP_METHODS).toMatchSnapshot();
  });

  test('The SALT Rounds value is correct', () => {
    expect(constants.SALT_ROUNDS).toBe(10);
  });
});
