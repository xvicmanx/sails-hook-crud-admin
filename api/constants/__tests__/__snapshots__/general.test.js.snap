// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api::constants::general The CRUD model names are correct 1`] = `
Array [
  "crudaction",
  "crudresource",
  "crudright",
  "crudgroup",
  "cruduser",
  "crudasset",
]
`;

exports[`api::constants::general The HTTP methods are correct 1`] = `
Array [
  "GET",
  "POST",
  "PUT",
  "PATCH",
  "DELETE",
  "OPTIONS",
]
`;
